bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQPartitionHandler.java

```diff
@@ -0,0 +1,357 @@
+                        if (armingType == ArmingActionType.DISARM) {
+                            code = disarmCode;
+                        } else {
+                            code = armCode;
+                        }
+                    }
+                    panel.sendAction(new ArmingAction(armingType, partitionId(), code));
+                } catch (IllegalArgumentException e) {
+                    logger.debug("Unknown arm type {} to channel {}", armingTypeName, channelUID);
+                }
+            }
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(QolsysIQChildDiscoveryService.class);
+    }
+
+    @Override
+    public void setDiscoveryService(QolsysIQChildDiscoveryService service) {
+        this.discoveryService = service;
+    }
+
+    @Override
+    public void startDiscovery() {
+        refresh();
+    }
+
+    /**
+     * The partition id
+     *
+     * @return
+     */
+    public int partitionId() {
+        return partitionId;
+    }
+
+    public void zoneActiveEvent(ZoneActiveEvent event) {
+        QolsysIQZoneHandler handler = zoneHandler(event.zone.zoneId);
+        if (handler != null) {
+            handler.zoneActiveEvent(event);
+        }
+    }
+
+    public void zoneUpdateEvent(ZoneUpdateEvent event) {
+        QolsysIQZoneHandler handler = zoneHandler(event.zone.zoneId);
+        if (handler != null) {
+            handler.zoneUpdateEvent(event);
+        }
+    }
+
+    protected void alarmEvent(AlarmEvent event) {
+        // is this needed? I need to trigger the alarm and see if two updates are sent
+        if (event.alarmType != AlarmType.NONE && event.alarmType != AlarmType.ZONEOPEN) {
+            updatePartitionStatus(PartitionStatus.ALARM);
+        }
+        updateAlarmState(event.alarmType);
+    }
+
+    protected void armingEvent(ArmingEvent event) {
+        updatePartitionStatus(event.armingType);
+        updateDelay(event.delay == null ? 0 : event.delay);
+        Partition partitionCache = this.partitionCache;
+        if (partitionCache != null) {
+            partitionCache.status = event.armingType;
+        }
+    }
+
+    protected void errorEvent(ErrorEvent event) {
+        cancelErrorDelayJob();
+        updateState(QolsysIQBindingConstants.CHANNEL_PARTITION_ERROR_EVENT, new StringType(event.description));
+        errorFuture = scheduler.schedule(this::clearErrorEvent, CLEAR_ERROR_MESSSAGE_TIME, TimeUnit.SECONDS);
+    }
+
+    protected void secureArmInfoEvent(SecureArmInfoEvent event) {
+        setSecureArm(event.value);
+    }
+
+    public void zoneAddEvent(ZoneAddEvent event) {
+        discoverZone(event.zone);
+    }
+
+    protected void updatePartition(Partition partition) {
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        this.partitionCache = partition;
```

Comment:
`partitionCache` is apparently defined/set but never used. Is it useful?

Reply:
not useful, left over from early prototype